renderer/src/common/components/settings/registry/registry-source-field.tsx

```diff
@@ -25,58 +76,34 @@ export function RegistrySourceField({
     return null
   }
 
+  const config =
+    REGISTRY_SOURCE_CONFIG[registryType as keyof typeof REGISTRY_SOURCE_CONFIG]
+
   return (
     <FormField
       control={form.control}
       name="source"
       render={({ field }) => {
-        const isRemote = registryType === 'url'
-
         return (
           <FormItem className="w-full">
-            <FormLabel required>
-              {isRemote ? 'Registry URL' : 'Registry File Path'}
-            </FormLabel>
-            <FormDescription>
-              {isRemote ? (
-                <>
-                  Provide the HTTPS url of a remote registry (
-                  <Button
-                    asChild
-                    variant="link"
-                    size="sm"
-                    className="h-auto p-0"
-                  >
-                    <a
-                      href="https://raw.githubusercontent.com/stacklok/toolhive/refs/heads/main/pkg/registry/data/registry.json"
-                      target="_blank"
-                      rel="noopener noreferrer"
-                    >
-                      official ToolHive registry
-                    </a>
-                  </Button>
-                  ).
-                </>
-              ) : (
-                'Provide the absolute path to a local registry JSON file on your system.'
-              )}
-            </FormDescription>
+            <FormLabel required>{config.label}</FormLabel>
+            <FormDescription>{renderDescription(registryType)}</FormDescription>
             <FormControl>
-              {isRemote ? (
-                <Input
-                  placeholder={'https://domain.com/registry.json'}
-                  {...field}
-                  disabled={isPending}
-                />
-              ) : (
+              {config.useFilePicker ? (
```

Comment:
not really related to your changes but I just noticed that we could probably extract this branching into a separate component like 

```tsx
function RegistryFormControl(...) {
  if (useFilePicker) {
     return <FilePickerInput />
  }

  return <Input />
}
```f